src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -86,6 +87,50 @@ public ProjectFactory(string msbuildDirectory, string path, IDictionary<string,
             Initialize(project);
         }
 
+        // This handler is called only when the common language runtime tries to bind to the assembly and fails
+        private Assembly AssemblyResolve(object sender, ResolveEventArgs args)
+        {
+            if (string.IsNullOrEmpty(_msbuildDirectory))
+            {
+                return null;
+            }
+
+            var failingAssemblyFilename = args.Name.Substring(0, args.Name.IndexOf(","));
+
+            // If we're failing to load a resource assembly, we need to find it in the appropriate subdir
+            if (failingAssemblyFilename.Length > 10 &&
+                failingAssemblyFilename.Substring(failingAssemblyFilename.Length - 10, 10).Equals(".resources", StringComparison.OrdinalIgnoreCase))
+            {
+                var fallBackToEnglish = false;
+                var cultureName = CultureInfo.CurrentCulture?.TwoLetterISOLanguageName;
+                if (string.IsNullOrEmpty(cultureName))
+                {
+                    fallBackToEnglish = true;
+                }
+
+                var resourceDir = Path.Combine(_msbuildDirectory, cultureName);
+                if (!Directory.Exists(resourceDir))
```

Comment:
This check isn't needed if this is already english.

You could reduce a lot of this code here by constructing the full paths for both options, and at the end return the first one that exists.